### PR TITLE
perf(b-table): cache cell slot names each render cycle (addresses #4008)

### LINKED ISSUE #4008: <b-table>: Select performance issue
### Describe the bug

select rows on a large btable introduce a ton of cell re-rendering even for unrelated rows, resulting in poor performance.

### Steps to reproduce the bug

1. Go to https://jsfiddle.net/wk8zd50e/
2. Click on one or more rows
3. See the sloppiness of the select update 

### Expected behavior

I would expect the selected row to be shown instantly, I would expect to see in the vue dev-tools the re-rendering of the affected rows only, not the whole table

### Versions

**Libraries:**

- BootstrapVue: 2.0.0-rc.28 (latest)
- Bootstrap: 4.3.1
- Vue: 2.6.10

**Environment:**

- OS: Windows
- Browser: Chrome, Firefox
- Version: latest

### PR DESCRIPTION
**Pull request by tmorehouse**
### Describe the PR

Will help improve performance (slightly) by not having to compute the possible slot names for each row rendered.

The cell slot names will only be looked-up once each render cycle.

Uses a non-reactive property on the table instance to store the cache.

Closes #4008 (partially)

Performance is still slow when table has 1000+ rows displayed, but is slightly improved.

User should be presenting paginated data (e.g. 100 rows at a time) rather than generating 1000 rows on screen.

### PR checklist

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [ ] Feature
- [ ] Enhancement
- [ ] ARIA accessibility
- [ ] Documentation update
- [x] Other (please describe) performance

**Does this PR introduce a breaking change?** (check one)

- [x] No
- [ ] Yes (please describe)

**The PR fulfills these requirements:**

- [x] It's submitted to the `dev` branch, **not** the `master` branch
- [x] When resolving a specific issue, it's referenced in the PR's title (i.e. `[...] (fixes #xxx[,#xxx])`, where "xxx" is the issue number)
- [x] It should address only one issue or feature. If adding multiple features or fixing a bug and adding a new feature, break them into separate PRs if at all possible.
- [x] The title should follow the [**Conventional Commits**](https://www.conventionalcommits.org/) naming convention (i.e. `fix(alert): not alerting during SSR render`, `docs(badge): update pill examples, fix typos`, `chore: fix typo in README`, etc). **This is very important, as the `CHANGELOG` is generated from these messages.**

**If new features/enhancement/fixes are added or changed:**

- [ ] Includes documentation updates (including updating the component's `package.json` for slot and event changes)
- [ ] Includes any needed TypeScript declaration file updates
- [ ] New/updated tests are included and passing (if required)
- [x] Existing test suites are passing
- [x] The changes have not impacted the functionality of other components or directives
- [ ] ARIA Accessibility has been taken into consideration (Does it affect screen reader users or keyboard only users? Clickable items should be in the tab index, etc.)

**If adding a new feature, or changing the functionality of an existing feature, the PR's
description above includes:**

- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

### CODE REVIEW

File: src/components/table/helpers/mixin-tbody-row.js [hunk 1]
```diff
@@ -205,10 +205,11 @@ export default {
       // v-slot attributes are lower-cased by the browser.
       // Switched to round bracket syntax to prevent confusion with
       // dynamic slot name syntax.
-      const slotNames = [`cell(${key})`, `cell(${key.toLowerCase()})`, 'cell()']
-      let $childNodes = this.hasNormalizedSlot(slotNames)
-        ? this.normalizeSlot(slotNames, slotScope)
-        : toString(formatted)
+      // const slotNames = [`cell(${key})`, `cell(${key.toLowerCase()})`, 'cell()']
```
Comment (jackmu95) [inline src/components/table/helpers/mixin-tbody-row.js]: @tmorehouse What about this commented out line?

Comment (tmorehouse) [reply src/components/table/helpers/mixin-tbody-row.js]: I just left it in for reference to the slot names we look for.

Comment (tmorehouse) [reply src/components/table/helpers/mixin-tbody-row.js]: Cold be re-worded as a true comment

Comment (tmorehouse) [reply src/components/table/helpers/mixin-tbody-row.js]: Changed it to a descriptive comment

**Pull request merged**